### PR TITLE
Added proposal and quick fix to remove annotations from parameters

### LINKED ISSUE #212: Dependency Injection: Quickfix for bean constructor with parameter anotated with `@Disposes`, `@Observes` or `@ObservesAsync`
[Declaring a Bean Constructor](https://jakarta.ee/specifications/cdi/3.0/jakarta-cdi-spec-3.0.html#declaring_bean_constructor)

> If a bean constructor has a parameter annotated `@Disposes`, `@Observes`, or `@ObservesAsync`, the container automatically detects the problem and treats it as a definition error.

Diagnostics for delivering error already exists and is shown at the constructor.

## Quickfixes
- Remove the annotation

## Example
```
@ApplicationScoped
public class Greeting{
	@Inject
	public Greeting (@ObservesAsync String string) {};
	
	public String greet(String name) {
        return "Greeting, " + name;
    }
}
```
```
[INFO] [ERROR   ] CWWKZ0004E: An exception occurred while starting the application demo-servlet. The exception message was: com.ibm.ws.container.service.state.StateChangeException: org.jboss.weld.exceptions.WeldException: WELD-000076: Bean constructor must not have a parameter annotated with @ObservesAsync: [EnhancedAnnotatedConstructorImpl] @Inject public io.openliberty.sample.jakarta.di.error.con.Greeting(@ObservesAsync String)
```

Related to #153

### PR DESCRIPTION
**Pull request by AlvinTan2000**
Fixes #212 and #105

### CODE REVIEW

**Comment by kathrynkodama** [comment]
See https://github.com/kathrynkodama/lsp4jakarta/commit/69cf6eddd436c188cab8b9c8b322d3d48fc17310
for a quick example solution of how we can modify an existing `RemoveModifierConflictQuickFix` class to remove annotations on parameters. 
Code can be cleaned up and refactored for resuability.

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/cdi/ManagedBeanTest.java [hunk 1]
```diff
@@ -234,6 +236,16 @@ public void producesAndDisposesObservesObservesAsync() throws Exception {
...
+        
+        
+        JakartaJavaCodeActionParams codeActionParams1 = createCodeActionParams(uri, d1);
+
+        TextEdit te1 = te(11, 4, 12, 4, "");
+        TextEdit te2 = te(12, 33, 13, 43, "");
+        CodeAction ca1 = ca(uri, "Remove @Inject", d1, te1);
+        CodeAction ca2 = ca(uri, "Remove the @Diposes modifier from this ", d1, te2);
+
+        assertJavaCodeAction(codeActionParams1, utils, ca1, ca2);
```
Comment (AlvinTan2000) [inline jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/cdi/ManagedBeanTest.java:248]: The test is failing here as it expects 4 code actions, namely:
- "Remove the `@Observes` modifier from this "
- "Remove the `@ObservesAsync` from this "

Comment (kathrynkodama) [reply jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/cdi/ManagedBeanTest.java:248]: and how many code actions is it returning? What is the actual vs expected?

Comment (AlvinTan2000) [reply jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/cdi/ManagedBeanTest.java:248]: Sorry for the confusion, I am expecting 2 code actions:
-  "Remove @Inject "
-  "Remove the @Disposes from this "

 But according to JUnit, the actual is 4.

Comment (kathrynkodama) [reply jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/cdi/ManagedBeanTest.java:248]: And are these the 4 code actions returned from JUnit?

- "Remove @Inject "
- "Remove the @Disposes from this "
- "Remove the @Observes modifier from this "
- "Remove the @ObservesAsync from this "

Comment (AlvinTan2000) [reply jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/cdi/ManagedBeanTest.java:248]: Yes

**Review by kathrynkodama — APPROVED** [review]
Looks good to me, thanks @AlvinTan2000!

**Pull request merged**